### ISSUE TITLE
Unable to resolve reference to 'some/path/README.md' for \ref command

### ISSUE THREAD

**Issue by godexsoft** [doxygen/doxygen#10906]
**Describe the bug**
Upgrading to 1.11.0 broke generation for some refs. Specifically in our mainpage it broke a link to a README.md. 
I'm getting the warning in the title now.

The project is opensource so here are some links:
- mainpage doc generated with previous doxygen version: https://xrplf.github.io/clio/ (at the bottom, `rpc/README.md` link)
- mainpage src: https://github.com/XRPLF/clio/blob/develop/src/main/Mainpage.hpp

Note that we are using the `[text](link)` syntax but the error happens for any sort of `@ref` usage.

**Expected behavior**
It should work the same way it worked on previous version.

**Version**
1.11.0

**Comment by godexsoft** [comment]
> > Upgrading to 1.11.0
> 
> From which version?
From the previous release 1.10.0. 

> 
> The given link might not be persistent and also the Doxyfile is of interest so:
> 
> * Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

The links i posted are not going anywhere. I don't have time to create small self-contained projects, sorry.
The Doxyfile (at the time of writing this) is here: https://github.com/XRPLF/clio/blob/067dd72aed90c2649717b1d27e52352af1000b71/docs/Doxyfile (permalink)

**Comment by albert-github** [comment]
> The links i posted are not going anywhere.

probably they will when the account is removed (unlikely but ...)


In the linked Doxyfile I see constructs like `${LINT}` would be nice to know what the used values are and not be dependent on the build procedure of project.

**Comment by godexsoft** [comment]
> > The links i posted are not going anywhere.
> 
> probably they will when the account is removed (unlikely but ...)
> 
> In the linked Doxyfile I see constructs like `${LINT}` would be nice to know what the used values are and not be dependent on the build procedure of project.

I give up. Here is your small self-contained project. Just run `./check-docs.sh` to see it fail :rage4: 
[doxybug.tar.gz](https://github.com/doxygen/doxygen/files/15485768/doxybug.tar.gz)

**Comment by godexsoft** [comment]
That's my bad - should have reverted my experiments before copying the files. Glad you were able to reproduce 👍

**Comment by albert-github** [comment]
I did a bisection and git:
```
15e5493cc4e79ec44d55d1ef1f4d5d2bc08290f4 is the first bad commit

Date:   Sun May 5 13:21:58 2024 +0200

    Avoid double calls to resolve references

 src/docnode.cpp   | 18 ++++++------------
 src/docparser.cpp | 19 +++++++++++++------
 src/docparser.h   |  1 +
 src/docparser_p.h |  1 +
 src/util.cpp      | 50 +-------------------------------------------------
 src/util.h        |  3 ---
 6 files changed, 22 insertions(+), 70 deletions(-)
```

(e8bd0b3b5214fe8687408689b2dc6c1959ea0373is still OK).

I suspect that in the optimization a `.` is replaced by `::` ant thus the reference cannot be resolved anymore and the target input of `DocRef::DocRef` can be a filename.

As far as I can tell the line:
```
if (sec==nullptr && parser->context.lang==SrcLangExt::Markdown)
```
is there problem now, the old construct was:
```
  SrcLangExt lang = getLanguageFromFileName(target);
  if (sec==nullptr && lang==SrcLangExt::Markdown) // lookup as markdown file
```

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10908

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by godexsoft** [comment]
@albert-github do you have any insight on when this will be released? thanks! 🚀

**Comment by albert-github** [comment]
I have no idea. We recently had a release (May 20, 2024) and usually there are 2 or 3 releases in a year (so maybe end of August or around Christmas is my best guess).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
